drivers/QSPI.h

```diff
@@ -39,6 +41,10 @@ namespace mbed {
  * @{
  */
 
+/** Type representing a QSPI instruction
+ */
+typedef uint8_t qspi_inst_t;
```

Comment:
`qspi_status_t` in the HAL defines the instruction as `uint8_t`. However, the drivers use "fundamental types", see other drivers like SPI. We use C99 in HAL. See https://os.mbed.com/handbook/C-Data-Types for details.

We should fix the storage if they use unsigned for whatever reason to use the same one.

Reply:
Updated the typdef to use `char` instead.